maddy.conf

```diff
@@ -46,6 +46,9 @@ smtp tcp://0.0.0.0:25 {
         # Verify that domain in MAIL FROM does have a MX record.
         require_mx_record
 
+        # Verify that TLS or STARTTLS is being used by the client. Fail if it is not.
+        require_tls
```

Comment:
I am slightly worried about that being in the default configuration, though. I think some legitimate messages are still delivered over plain-text connections, I do [not want to break it][1]. To mitigate downgrade attacks however (if sender server lacks the DANE/DNSSEC or MTA-STS support or it is not configured by the recipient), maddy might remember the use of TLS by the sender server and enforce it second time (like chasquid [does][2]).

[1]: https://github.com/foxcpp/maddy/blob/master/HACKING.md#design-goals
[2]: https://blitiri.com.ar/p/chasquid/docs/sec-levels/

Reply:
Yeah, it probably doesn't want to be on by default. I was wondering where best to add it to the manpage.

Reply:
Perhaps add a "Connection state checks" section to maddy-filter(5) and then mention require_tls in a subsection?

Reply:
I've renamed "DNS checks", since it's fundamentally the same kind of thing. WDYT?

Reply:
(I figure we could add tls preference memory in a later PR)